.github/workflows/ci.yml

```diff
@@ -138,5 +144,39 @@ jobs:
         run: tar -xvf ../rootfs-${{ matrix.target }}.tar
 
       - name: Test runner
+        id: runner_id
         run: |
           python3 ./phoenix-rtos-tests/runner.py -T${{ matrix.target }}
+
+  mail-notification:
+    if: ${{ failure() }}
+    needs: ['build', 'test-emu', 'test-hw']
+    name: notify ci team using e-mail
+    runs-on: ubuntu-latest
+    steps:
+      - name: Send mail
+        # If there is some failure, not caused by build step or test runner step send mail
+        if: >-
+          ${{ needs.build.outputs.output_build != 'failure'
+          && needs.test-emu.outputs.output_runner != 'failure'
+          && needs.test-hw.outputs.output_runner != 'failure' }}
+        uses: dawidd6/action-send-mail@v3
+        with:
+          # Required mail server address:
```

Comment:
please remove comments which are not introducing additional information. As I see, most of the parameters are self-descriptive